pkg/operator/loglevel/controller.go

```diff
@@ -0,0 +1,113 @@
+
+	log "github.com/sirupsen/logrus"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	logLevelUtils "github.com/openshift/library-go/pkg/operator/loglevel"
+
+	"github.com/openshift/cloud-credential-operator/pkg/operator/status"
+	"github.com/openshift/cloud-credential-operator/pkg/operator/utils"
+)
+
+const (
+	controllerName = "loglevel"
+)
+
+// Add creates a new ConfigController and adds it to the Manager.
+func Add(mgr manager.Manager, kubeConfig string) error {
+	r := newReconciler(mgr)
+
+	// Create a new controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to CloudCredential object and reconcile the loglevel changes
+	err = c.Watch(
+		&source.Kind{Type: &operatorv1.CloudCredential{}},
+		&handler.EnqueueRequestForObject{})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	r := &ReconcileCloudCredentialConfig{
+		Client: mgr.GetClient(),
+	}
+	status.AddHandler(controllerName, r)
+
+	return r
+}
+
+var _ reconcile.Reconciler = &ReconcileCloudCredentialConfig{}
+
+// ReconcileCloudCredentialConfig reconciles a CredentialConfig object
+type ReconcileCloudCredentialConfig struct {
+	client.Client
+}
+
+// Reconcile reads the state of the cluster for a CredentialConfig object and
+// makes changes based on the state read and what is in the CredentialConfig.LogLevel
+// Automatically generate RBAC rules to allow the Controller to read and write required types.
+// +kubebuilder:rbac:groups=operator.openshift.io,resources=cloudcredential/spec,verbs=get
+func (r *ReconcileCloudCredentialConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	logger := log.WithFields(log.Fields{
+		"controller": controllerName,
+		"cr":         fmt.Sprintf("%s/%s", request.NamespacedName.Namespace, request.NamespacedName.Name),
+	})
+
+	// GetOperatorLogLevel
+	log.Debug("Checking log level for Reconcile request")
+	desiredLogLevel, err := utils.GetOperatorLogLevel(r, logger)
+
+	// Error with the current logLevel
+	if err != nil || !logLevelUtils.ValidLogLevel(desiredLogLevel) {
+		return reconcile.Result{Requeue: false}, fmt.Errorf("the loglevel defined for the cloudcredential object is invalid")
+	}
+
+	currentLogLevel, errBool := logLevelUtils.GetLogLevel()
+	if errBool {
+		log.Debug("Unable to get the current loglevel. Defaulting to 'Normal'")
+	}
+
+	if currentLogLevel == desiredLogLevel {
+		return reconcile.Result{Requeue: false}, nil
+	}
+
+	// Set logLevel to the new level
+	logLevelUtils.SetLogLEvel(desiredLogLevel)
```

Comment:
I think we need code to set both `logLevel` and `operatorLogLevel`

Reply:
Yeah, I agree, but I wasn't sure what to set when logLevel is changed, as this (AFAIK) Operator doesn't have managed components but instead has the actuators in the Operator.

Would you prefer if both operatorLogLevel and logLevel define the Operator loglevel and the highest verbosity wins?

Reply:
> Would you prefer if both operatorLogLevel and logLevel define the Operator loglevel and the highest verbosity wins?

Yes, that sounds good to me